src/js_native_api_v8.h

```diff
@@ -55,17 +55,15 @@ struct napi_env__ {
     CHECK_EQ(isolate, context->GetIsolate());
     napi_clear_last_error(this);
   }
-  virtual ~napi_env__() { FinalizeAll(); }
-  v8::Isolate* const isolate;  // Shortcut for context()->GetIsolate()
-  v8impl::Persistent<v8::Context> context_persistent;
+  virtual ~napi_env__() = default;
```

Comment:
Maybe try if you can make the constructor protected, to avoid callers deleting it without proper cleanup.

Reply:
Make sense to me. Thank you for the suggestion! Updated.